geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/SizeableObject2ObjectOpenCustomHashMapWithCursor.java

```diff
@@ -29,52 +33,58 @@
  * The scan method provides the same guarantees as Redis's HSCAN, and in fact
  * uses the same algorithm.
  */
-public class Object2ObjectOpenCustomHashMapWithCursor<K, V>
-    extends Object2ObjectOpenCustomHashMap<K, V> {
+public class SizeableObject2ObjectOpenCustomHashMapWithCursor<K, V>
+    extends Object2ObjectOpenCustomHashMap<K, V> implements Sizeable {
+
+  private static final long serialVersionUID = 9079713776660851891L;
+  public static final int BACKING_ARRAY_OVERHEAD_CONSTANT = 136;
+  public static final int BACKING_ARRAY_LENGTH_COEFFICIENT = 4;
+  private static final ReflectionSingleObjectSizer elementSizer =
+      ReflectionSingleObjectSizer.getInstance();
 
+  private int keysOverhead;
```

Comment:
Currently these "Overhead" ints only include the size of the byte arrays. But if every time you changed the Overhead for a byte array you also included the BACKING_ARRAY_LENGTH_COEFFICIENT (by incing it or decing it) then you would not need to consult the length the arrays later when you compute the total size. Seems like it would make it a little faster.

Reply:
The length of the arrays isn't updated every time an element is added to the collection, only when there are sufficient elements to fill a percentage of it determined by the load factor, which then forces a resize of the backing arrays, so it's necessary to check the length of the arrays to determine the current overhead.